src/Controller/FacebookAuthController.php

```diff
@@ -145,32 +168,63 @@ public function returnFromFb() {
       return $this->redirect('user.login');
     }
 
-    $this->facebookManager->setClient($facebook)->authenticate();
+    $state = $this->persistentDataHandler->get('oAuth2State');
 
-    // Checks that user authorized our app to access user's email address.
-    if (!$this->facebookManager->checkPermission('email')) {
-      drupal_set_message($this->t('Facebook login failed. This site requires permission to get your email address from Facebook. Please try again.'), 'error');
-      $this->persistentDataHandler->set('reprompt', TRUE);
+    if (empty($_GET['state']) || ($_GET['state'] !== $state)) {
+      unset($_SESSION['oauth2state']);
```

Comment:
You shouldn't use $_SESSION variable directly, instead use Symfony session service